controller/work_item_comments_blackbox_test.go

```diff
@@ -125,6 +126,37 @@ func assertWorkItemComment(t *testing.T, c *app.Comment, expectedBody string, ex
 	assert.NotNil(t, c.Relationships.Creator.Data.ID)
 }
 
+func (rest *TestCommentREST) TestSuccessCreateSingleCommentWithParentComment() {
+	// given
+	wi := rest.createDefaultWorkItem()
+
+	// create parent
+	markup := rendering.SystemMarkupMarkdown
+	p := rest.newCreateWorkItemCommentsPayload("Test", &markup)
+	svc, ctrl := rest.SecuredController()
+	_, c := test.CreateWorkItemCommentsOK(rest.T(), svc.Context, svc, ctrl, wi.ID, p)
+	// then
+	assertComment(rest.T(), c.Data, rest.testIdentity, "Test", markup)
+
+	// create child
+	parentID := c.Data.ID.String()
+	child := rest.newCreateWorkItemCommentsPayload("Test Child", &markup)
+	child.Data.Relationships = &app.CreateCommentRelations{
+		ParentComment: &app.RelationGeneric{
+			Data: &app.GenericData{
+				Type: ptr.String("comments"),
+				ID:   ptr.String(c.Data.ID.String()),
+			},
+		},
+	}
+	// make sure the above correctly sets the parentCommentID
+	assert.Equal(rest.T(), *child.Data.Relationships.ParentComment.Data.ID, parentID)
+	// exec create child
+	_, d := test.CreateWorkItemCommentsOK(rest.T(), svc.Context, svc, ctrl, wi.ID, child)
```

Comment:
You can use golden files to validate the entire response.

Reply:
@jarifibrahim All these tests need to be changed to use golden files. But do we need in this PR?

Reply:
@baijum Let's fix one test at a time. The older tests will be updated eventually, let's make sure the new tests are using golden files.

Reply:
No. All other tests are not using golden files either here. We need to get this in by this weekend, so we should not start to refactor existing code. We can and should do that later.

Reply:
Okay. Since this file requires major refactoring, let's do it separately.